Custom user filters

deanmarcussen opened the pull request:
Fixes https://github.com/OrchardCMS/OrchardCore/issues/7868

Extends the users admin list, in a similar way to the contents list, to provide filtering, and extensible filters (i.e. add your own, based on your own user profile data / indexes you've created).

TODO: Some clean up of old / never implemented filters

deanmarcussen commented on src/OrchardCore/OrchardCore.Users.Core/OrchardCore.Users.Core.csproj:
```diff
@@ -15,6 +15,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\OrchardCore.DisplayManagement.Abstractions\OrchardCore.DisplayManagement.Abstractions.csproj" />
```
Need this reference to have `IUpdateModel` 

Need `IUsersAdminListFilter` in the Core project because `User` is in the core project.

Wonder about moving `User` to the abstractions project, to avoid this `DisplayManagement.Abstractions` reference.

agriffard commented on src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs:
```diff
@@ -154,46 +133,71 @@ public async Task<ActionResult> Index(UserIndexOptions options, PagerParameters
-                new SelectListItem() { Text = S["All"], Value = nameof(UsersFilter.All) },
+                new SelectListItem() { Text = S["Enabled Users"], Value = nameof(UsersFilter.Enabled) },
+                new SelectListItem() { Text = S["Disabled Users"], Value = nameof(UsersFilter.Disabled) },
+                new SelectListItem() { Text = S["All Users"], Value = nameof(UsersFilter.All) },
```
Suggestion: Move `All Users` at the top of the list.

deanmarcussen commented:
Moved to including `Authenticated` in the roles index (as the equivalent of `No roles`).

No migration for this, updated when editing user (don't want to do yet another user migration for it)

UI now shows `Authenticated (no roles)` as the text for this filter.

Leaving commented filters (never implemented) in the code base, maybe someone will take it as a hint to see about considering implementations for them
